lib/ansible/constants.py

```diff
@@ -36,6 +40,26 @@ def _deprecated(msg, version):
         sys.stderr.write(' [DEPRECATED] %s, to be removed in %s\n' % (msg, version))
 
 
+def handle_config_noise(display=None):
+
+    if display is not None:
+        w = display.warning
+        d = display.deprecated
+    else:
+        w = _warning
```

Comment:
Especially if the pre-display fallback gets dropped, I'm not sure the microperf of avoiding the method lookup in the loop is worth the indirection here... Just how many warnings are you expecting to fire here? :wink:

Reply:
I expect 0, but its not always about what i expect